Mcj 06

ThomassenCathleia opened the pull request:
The basic model structure of this project

ThomassenCathleia committed:
Chore: first commit. This commit has the foundations of the project's
model layer.

ThomassenCathleia committed:
Renamed MenuItem to OrderItem

Changed the menuItems type in Menu to Product

Commit:
Added a receipt class

lenshuygh commented on src/be/mcjava/model/Ingredient.java:
```diff
@@ -0,0 +1,39 @@
+    }
+    
+    public boolean isAvailable() {
+        //todo: implement this, returns false so code will compile -Michiel 24/5/2019
```
isn't this already implementable now?
i would return (currentStock > 0); here

lenshuygh commented on src/be/mcjava/model/CustomerOrder.java:
```diff
@@ -0,0 +1,56 @@
+    }
+    
+    public void addToOrder(OrderItem orderItem) {
+    }
```
implementation can be done now with:
this.orderedItems.add(orderItem);
?

lenshuygh commented on src/be/mcjava/model/CustomerOrder.java:
```diff
@@ -0,0 +1,56 @@
+    }
+    
+    public void removeFromOrder(OrderItem orderItem) {
+    }
```
implement;
this.orderedItems.remove(orderItem);
?

lenshuygh commented on src/be/mcjava/model/Product.java:
```diff
@@ -0,0 +1,42 @@
+    }
+    
+    public boolean isAvailable() {
+        //todo: implement this, returns false so code will compile -Michiel 24/5/2019
```
implementation?
for (Ingredient ingredient : ingredients.keySet()) {
            if(ingredient.getCurrentStock() < ingredients.get(ingredient)){
                return false;
            }
        }
        return true;

lenshuygh commented:
Great work Michiel, thats a lot of work done!
Added a few comments on the helper-methods that can already be implemented...

ThomassenCathleia commented:
The main reason why I did not implement anything seriously was to just lay down the structure.

The boolean return methods are rather simple. The add/remove methods might need some better implementation, such as needing a custom equals and compare methods.

lenshuygh commented:
I get what you mean, but now they'll need to be implemented by making another story , if the methods are available and they don't do anything it will create confusion, especially since some do return something thats not a real result (eg: return false;)
If the method is available to other code you can't expect the person working on another story to go and check all your methods...
You can merge it as is but you'll need to add another story to make the implementations

ThomassenCathleia committed:
See MCJ_06 in McJavaKitchen for full details

Summary:
- added constructor to every class
- implemented all methods with business logic

ThomassenCathleia committed:
Merge remote-tracking branch 'origin/MCJ_06' into MCJ_06

lenshuygh reviewed (approved):
good stuff :-)
